lib/src/value/calculation.dart

```diff
@@ -287,6 +764,19 @@ class SassCalculation extends Value {
         "${pluralize('was', args.length, plural: 'were')} passed.");
   }
 
+  /// Returns a [Callable] named [name] that calls a single argument
+  /// math function.
+  static Value _singleArgument(String name, Object argument,
+      SassNumber mathFunc(SassNumber value), bool acceptsKnownUnits) {
+    argument = _simplify(argument);
+    if (argument is! SassNumber ||
+        (acceptsKnownUnits && argument.hasUnit('%'))) {
+      return SassCalculation._(name, [argument]);
+    }
+    if (!acceptsKnownUnits) argument.assertNoUnits();
```

Comment:
If a function requires no units, that's a different thing than whether it requires known units. Even if they happen to be completely complementary for the functions we use this helper for, they should be separate boolean parameters.

Reply:
Right, I added the these parameters: `{bool requireUnits = true, bool requireKnownUnits = false}`.